### ISSUE TITLE
Image Example

### ISSUE THREAD

**Issue by pavelxx1** [Kolkir/Coarse_LoFTR_TRT#1]
HI
Can u provide end-to-end example for image1<->image2 matching
thx

**Comment by Kolkir** [comment]
Hi @pavelxx1 Could you clarify what example do you need? The [webcam.py](https://github.com/Kolkir/Coarse_LoFTR_TRT/blob/main/webcam.py) demo shows how to initialize models, load weights and do the feature matching. Or maybe you require some video/screenshot example?

**Comment by pavelxx1** [comment]
`Or maybe you require some video/screenshot example?` 
Hi
Yes, example for two images from folder :)

**Comment by Kolkir** [comment]
@pavelxx1 I can provide you such example. But what do you expect as output? The model now outputs the following data(ndarrays):
[first image 2D coordinates array], [second image 2D coordinates array], [feature matching confidence level]
So should  the example just print them out to the console window? Or you need some graphical output(for example, stacked images with rendered correspondences, color will show the confidence level, or just threshold the best ones)?

**Comment by pavelxx1** [comment]
Thx
I want to compare result your model vs model from original loftr

**Comment by Kolkir** [comment]
@pavelxx1 Sorry, but it still is not clear what do you want. Could you provide an output file format description that will be suitable for your needs? Also, please take into account that my model doesn't do fine processing and matching, so there will a difference in the output pixel coordinates, and you will be not able to directly compare results.

**Comment by pavelxx1** [comment]
Uhh
I mean can u make .ipynb example such this _`/zju3dv/LoFTR/blob/master/notebooks/demo_single_pair.ipynb`_
but with your weights
thx

**Comment by pavelxx1** [comment]
just fixed your webcam code for colab compatibility :) `(single_pair)`
 ```
    a = '/content//faces/real/real_00001.jpg'
    im_a = cv2.imread(a)

    b = '/content//faces/real/real_00002.jpg'
    im_b = cv2.imread(b)

    img0 = torch.from_numpy(make_query_image(im_a,img_size))[None][None].to(device=device) / 255.0
    img1 = torch.from_numpy(make_query_image(im_b,img_size))[None][None].to(device=device) / 255.0

    # Inference with LoFTR and get prediction

    with torch.no_grad():
        conf_matrix, _ = matcher(img0, img1)
        conf_matrix = conf_matrix.cpu().numpy()

        mkpts0, mkpts1, mconf = get_coarse_match(conf_matrix, img_size[1], img_size[0], loftr_coarse_resolution)

        # filter only the most confident features
        n_top = 20
        indices = np.argsort(mconf)[::-1]
        indices = indices[:n_top]
        mkpts0 = mkpts0[indices, :]
        mkpts1 = mkpts1[indices, :]


        draw_features(im_a, mkpts0, img_size, color=(0, 255, 0))
        draw_features(im_b, mkpts1, img_size, color=(0, 255, 0))

        # combine images
        res_img = np.hstack((im_a, im_b))
        cv2.imwrite('/content/matches.png', res_img)
```

closed issue.

**Comment by pavelxx1** [comment]
I have another question:
1)for training did u use BlendedMVS low-res-dataset ?
2)how to create my own dataset. I have only images **w/o** any camera perspectives,depth,maps,mask etc for it
thx

**Comment by Kolkir** [comment]
1. Yes, I've used the BlendedMVS low-res.
2. In general, training procedure requires only images and feature correspondences. I've used depth maps and camera matrices  just to extract feature matches and build a confidence map. So if you have feature matches for your images, it should be easy to change the training code, I can help you.
